--- graf3d/ftgl/CMakeLists.txt	2020-03-13 09:13:48.351131291 +0100
+++ graf3d/ftgl/CMakeLists.txt_new	2020-03-13 09:14:49.584310798 +0100
@@ -2,7 +2,7 @@
 # CMakeLists.txt file for building ROOT graf3d/ftgl package
 ############################################################################
 
-include_directories(${FREETYPE_INCLUDE_DIRS})
+include_directories(${FREETYPE_INCLUDE_DIRS} ${GLU_INCLUDE_DIR})
 
 if(MACOSX_GLU_DEPRECATED)
   add_definitions(-Wno-deprecated-declarations)
