--- FairLogger-1.2.0/CMakeLists.txt	2019-05-14 17:10:27.200976327 +0200
+++ FairLogger-1.2.0/CMakeLists.txt	2019-05-14 17:10:42.960611377 +0200
@@ -14,7 +14,7 @@
 include(FairLoggerLib)
 
 set_fairlogger_cmake_policies()
-get_git_version(OUTVAR_PREFIX FairLogger)
+get_git_version(OUTVAR_PREFIX FairLogger DEFAULT_VERSION 1.2.0)
 
 project(FairLogger VERSION ${FairLogger_VERSION} LANGUAGES C CXX)
 message(STATUS "${BWhite}${PROJECT_NAME}${CR} ${FairLogger_GIT_VERSION} from ${FairLogger_DATE}")
