--- CMakeLists.txt	2019-10-29 11:22:24.000000000 +0100
+++ CMakeLists.txt_new	2019-10-29 11:26:26.000000000 +0100
@@ -20,24 +20,26 @@
 # or take it from the version file if git is unavailable (a package distribution)
 # Keep the version file in the source directory to satisfy the source package.
 set( DDS_VERSION_FILE ${CMAKE_CURRENT_BINARY_DIR}/version )
-if( EXISTS "${CMAKE_SOURCE_DIR}/.git" )
+#if( EXISTS "${CMAKE_SOURCE_DIR}/.git" )
 
 	# The command produce a verion like "X.Y.Z.gHASH"
 	# The hash suffix is "-g" + 7-char abbreviation for the tip commit of parent.
 	# The "g" prefix stands for "git" and is used to allow describing the version of a software depending on the SCM the software is managed with.
-  execute_process(COMMAND git --git-dir "${CMAKE_SOURCE_DIR}/.git" describe --match "[0-9]*\\.[0-9]*" --abbrev=7 HEAD
-                  COMMAND sed -e "s/-/./g"
-                  OUTPUT_VARIABLE DDS_VERSION
-                  OUTPUT_STRIP_TRAILING_WHITESPACE)
-  execute_process( COMMAND bash -c "echo \"${DDS_VERSION}\" > ${DDS_VERSION_FILE}" )
-
-else( EXISTS "${CMAKE_SOURCE_DIR}/.git" )
-
-  execute_process(COMMAND cat ${DDS_VERSION_FILE}
-                  OUTPUT_VARIABLE DDS_VERSION
-                  OUTPUT_STRIP_TRAILING_WHITESPACE)
-
-endif( EXISTS "${CMAKE_SOURCE_DIR}/.git" )
+#  execute_process(COMMAND git --git-dir "${CMAKE_SOURCE_DIR}/.git" describe --match "[0-9]*\\.[0-9]*" --abbrev=7 HEAD
+#                  COMMAND sed -e "s/-/./g"
+#                  OUTPUT_VARIABLE DDS_VERSION
+#                  OUTPUT_STRIP_TRAILING_WHITESPACE)
+#  execute_process( COMMAND bash -c "echo \"${DDS_VERSION}\" > ${DDS_VERSION_FILE}" )
+
+#else( EXISTS "${CMAKE_SOURCE_DIR}/.git" )
+
+#  execute_process(COMMAND cat ${DDS_VERSION_FILE}
+#                  OUTPUT_VARIABLE DDS_VERSION
+#                  OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+#endif( EXISTS "${CMAKE_SOURCE_DIR}/.git" )
+File(WRITE ${DDS_VERSION_FILE} "2.5")
+Set(DDS_VERSION 2.5)
 message(STATUS "Building DDS version: ${DDS_VERSION}")
 #
 # Options
