--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -381,6 +381,14 @@
-# because of SIP on macOS we can't use (DY)LD_LIBRARY_PATH.
-# But we need to search also in custom location for libstdc++ in case if user installs a custom version of gcc/clang.
-# WORKAROUND: We therefore introduce DDS_LD_LIBRARY_PATH, which user can use to specify custom library path(s)
-set(ENV_LD_LIBRARY_PATH "$ENV{DDS_LD_LIBRARY_PATH}")
+if(ENV{DDS_LD_LIBRARY_PATH})
+  # because of SIP on macOS we can't use (DY)LD_LIBRARY_PATH.
+  # But we need to search also in custom location for libstdc++ in case if user installs a custom version of gcc/clang.
+  # WORKAROUND: We therefore introduce DDS_LD_LIBRARY_PATH, which user can use to specify custom library path(s)
+  file(TO_CMAKE_PATH "$ENV{DDS_LD_LIBRARY_PATH}" ENV_LD_LIBRARY_PATH)
+else()
+  file(TO_CMAKE_PATH "$ENV{LD_LIBRARY_PATH}" ENV_LD_LIBRARY_PATH)
+endif()
+
+set(PREREQ_DIRS "$<TARGET_FILE_DIR:dds-user-defaults>::$<TARGET_FILE_DIR:dds_protocol_lib>::$<TARGET_FILE_DIR:dds_intercom_lib>::$<TARGET_FILE_DIR:dds_topology_lib>::$<TARGET_FILE_DIR:dds_ncf>::$<TARGET_FILE_DIR:Boost::log>")
+foreach(p IN LISTS ENV_LD_LIBRARY_PATH)
+  set(PREREQ_DIRS "${PREREQ_DIRS}::${p}")
+endforeach()

-set(PREREQ_DIRS "$<TARGET_FILE_DIR:dds-user-defaults>::$<TARGET_FILE_DIR:dds_protocol_lib>::$<TARGET_FILE_DIR:dds_intercom_lib>::$<TARGET_FILE_DIR:dds_topology_lib>::$<TARGET_FILE_DIR:dds_ncf>::${DDS_BOOST_LIB_DIR}::${ENV_LD_LIBRARY_PATH}")
