commit 35c7959c530c010da718892fd3948467f104d267
Author: Alexey Rybalchenko <alexryba@gmail.com>
Date:   Mon Sep 21 18:48:27 2020 +0200

    Workaround Cpp17MoveInsertable issue on xcode 12

diff --git a/fairmq/FairMQChannel.h b/fairmq/FairMQChannel.h
index ba6ea31c..f472a99c 100644
--- a/fairmq/FairMQChannel.h
+++ b/fairmq/FairMQChannel.h
@@ -66,13 +66,13 @@ class FairMQChannel
     FairMQChannel(const FairMQChannel&, const std::string& name);
 
     /// Move constructor
-    FairMQChannel(FairMQChannel&&) = delete;
+    // FairMQChannel(FairMQChannel&&) = delete;
 
     /// Assignment operator
     FairMQChannel& operator=(const FairMQChannel&);
 
     /// Move assignment operator
-    FairMQChannel& operator=(FairMQChannel&&) = delete;
+    // FairMQChannel& operator=(FairMQChannel&&) = delete;
 
     /// Destructor
     virtual ~FairMQChannel()
