--- cmake/Modules/Geant4MakeRules_cxx.cmake_orig	2015-11-17 11:28:34.000000000 +0100
+++ cmake/Modules/Geant4MakeRules_cxx.cmake	2015-11-17 11:29:14.000000000 +0100
@@ -86,2 +86,2 @@
-  if(APPLE AND "${_clangcxx_version}" MATCHES ".*Apple LLVM version.*")
-    string(REGEX REPLACE ".*based on LLVM ([0-9](\\.[0-9])+)svn.*" "\\1" _clangcxx_version ${_clangcxx_version})
+  if(APPLE AND "${_clangcxx_version}" MATCHES ".* LLVM version.*")
+    string(REGEX REPLACE ".* LLVM version ([0-9][.0-9]+).*" "\\1" _clangcxx_version ${_clangcxx_version})
