--- pythia-6.4.28.f	2015-10-20 15:21:22.843710423 +0200
+++ generators/pythia6/pythia-6.4.28.f	2015-10-22 16:36:47.387713570 +0200
@@ -3198,7 +3198,10 @@
      &/PYINT1/,/PYINT2/,/PYINT4/,/PYINT5/
 C...Local array.
       DIMENSION VTX(4)
+c   count number of retries do to pyscat infinite loop
+      data nfail/0/
  
+ 55   continue
 C...Optionally let PYEVNW do the whole job.
       IF(MSTP(81).GE.20) THEN
         CALL PYEVNW
@@ -3289,7 +3292,13 @@
           MINT31=MINT(31)
   120     MINT(31)=MINT31
           MINT(51)=0
-          CALL PYSCAT
+          CALL PYSCAT(ifail)
+          if(ifail.gt.0) then
+            nfail=nfail+1
+            print *,'**********PSCAT*************',
+     +      ' in loop: nr of occurrences sofar=',nfail
+            goto 55
+          endif
           IF(MINT(51).EQ.1) GOTO 100
           IPU1=MINT(84)+1
           IPU2=MINT(84)+2
@@ -3672,7 +3681,7 @@
           MINT31=MINT(31)
   120     MINT(31)=MINT31
           MINT(51)=0
-          CALL PYSCAT
+          CALL PYSCAT(ifail)
           IF(MINT(51).EQ.1) GOTO 100
           NPARTD=N
           NFIN=N
@@ -4833,7 +4842,7 @@
         MINT31=MINT(31)
   110   MINT(31)=MINT31
         MINT(51)=0
-        CALL PYSCAT
+        CALL PYSCAT(ifail)
         IF(MINT(51).EQ.1) GOTO 100
         IPU1=MINT(84)+1
         IPU2=MINT(84)+2
@@ -10569,7 +10578,7 @@
 C...Finds outgoing flavours and event type; sets up the kinematics
 C...and colour flow of the hard scattering
  
-      SUBROUTINE PYSCAT
+      SUBROUTINE PYSCAT(ifail)
  
 C...Double precision and integer declarations
       IMPLICIT DOUBLE PRECISION(A-H, O-Z)
@@ -10626,7 +10635,8 @@
      & 5100012,5100011,5100014,5100013,5100016,5100015, 
      & 5100021,5100022,5100023,5100024/                 
       SAVE VINTSV
- 
+c  protection against looping around 180 
+      ifail=0 
 C...Read out process
       ISUB=MINT(1)
       ISUBSV=ISUB
@@ -10734,6 +10744,7 @@
   170 CONTINUE
  
 C...Choose new quark/lepton flavour for relevant annihilation graphs
+      n180=0
       IF(ISUB.EQ.12.OR.ISUB.EQ.53.OR.ISUB.EQ.54.OR.ISUB.EQ.58.OR.
      &ISUB.EQ.314.OR.ISUB.EQ.319.OR.ISUB.EQ.316.OR.
      &(ISUB.GE.135.AND.ISUB.LE.140).OR.ISUB.EQ.382.OR.ISUB.EQ.385) THEN
@@ -10741,6 +10752,7 @@
         IF(ISUB.EQ.58.OR.(ISUB.GE.137.AND.ISUB.LE.140)) IGLGA=22
         CALL PYWIDT(IGLGA,SH,WDTP,WDTE)
   180   RKFL=(WDTE(0,1)+WDTE(0,2)+WDTE(0,4))*PYR(0)
+        n180=n180+1
         DO 190 I=1,MDCY(IGLGA,3)
           KFLF=KFDP(I+MDCY(IGLGA,2)-1,1)
           RKFL=RKFL-(WDTE(I,1)+WDTE(I,2)+WDTE(I,4))
@@ -10749,7 +10761,15 @@
   200   CONTINUE
         IF((ISUB.EQ.53.OR.ISUB.EQ.385.OR.ISUB.EQ.314.OR.ISUB.EQ.319
      &      .OR.ISUB.EQ.316).AND.MINT(2).LE.2) THEN
-          IF(KFLF.GE.4) GOTO 180
+          IF(KFLF.GE.4) then
+            if(n180.gt.100) then
+              ifail=1
+              return
+            else
+              GOTO 180
+            endif
+          endif
+
         ELSEIF((ISUB.EQ.53.OR.ISUB.EQ.385.OR.ISUB.EQ.314.OR.ISUB.EQ.319.
      &       OR.ISUB.EQ.316).AND.MINT(2).LE.4) THEN
           KFLF=4
