diff --git a/gcalor/gcalor.F b/gcalor/gcalor.F
index 88962b6..05a0abd 100644
--- a/gcalor/gcalor.F
+++ b/gcalor/gcalor.F
@@ -15037,7 +15037,7 @@ C only p = 0, D = 7, T = 8, He3 = 9, alpha=10
      +                                0 , 7 , 8 , -1,
      +                               -1 ,-1 , 9 , 10/
       LOGICAL NOP
-      SAVE
+C      SAVE
 C first check, if ZEBRA still in order
       IF(LD(LMAG1).NE.NMAGIC.OR.LD(LMAG2).NE.NMAGIC) THEN
          WRITE(6,*) ' CALOR: ZEBRA banks screwed up --> STOP'
