--- a/grpc-proto/CMakeLists.txt
+++ b/grpc-proto/CMakeLists.txt
@@ -6,7 +6,7 @@
 set(odc_proto_file "odc.proto")

 # Generated sources
-protobuf_generate_cpp(odc_proto_srcs odc_proto_hdrs ${CMAKE_CURRENT_BINARY_DIR} ${odc_proto_file})
+protobuf_generate_cpp(odc_proto_srcs odc_proto_hdrs ${odc_proto_file})
 grpc_generate_cpp(odc_grpc_srcs odc_grpc_hdrs ${CMAKE_CURRENT_BINARY_DIR} ${odc_proto_file})

 add_library(odc_grpc_proto_lib STATIC
