--- CMakeLists.txt	2019-05-29 11:02:31.064729929 +0200
+++ CMakeLists.txt_new	2019-05-29 11:02:59.120076265 +0200
@@ -92,11 +92,11 @@
 # If alibuild is used the compiler flags are passed on the command line
 # and are also added to CMAKE_CXX_FLAGS.
 # So check if CMAKE_CXX_FLAGS has the compiler flags -std=c++11 or -std=c++14
-String(FIND ${CMAKE_CXX_FLAGS} "-std=c++11" POS_C++11)
+String(FIND "${CMAKE_CXX_FLAGS}" "-std=c++11" POS_C++11)
 If(${POS_C++11} EQUAL -1)
-  String(FIND ${CMAKE_CXX_FLAGS} "-std=c++14" POS_C++11)
+  String(FIND "${CMAKE_CXX_FLAGS}" "-std=c++14" POS_C++11)
   If(${POS_C++11} EQUAL -1)
-    String(FIND ${CMAKE_CXX_FLAGS} "-std=c++17" POS_C++11)
+    String(FIND "${CMAKE_CXX_FLAGS}" "-std=c++17" POS_C++11)
     If(${POS_C++11} EQUAL -1)
       Message(FATAL_ERROR "FairSoft wasn't compiled with c++11, c++14 or c++17 support. Please recompile FairSoft with a compiler which supports at least c++11.")
     EndIf()
