diff --git a/CMakeLists.txt b/CMakeLists.txt
index e8060c7d1d..f1074a3028 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -302,7 +307,7 @@ endif()
 get_cmake_property(variables CACHE_VARIABLES)
 foreach(var ${variables})
   if((var MATCHES "_(LIBRARIES|LIBRARY|INCLUDE|VERSION)") AND
-     (NOT ${${var}} STREQUAL "") AND
+     (NOT "${${var}}" STREQUAL "") AND
      (NOT ${var} MATCHES "NOTFOUND"))
     if (var MATCHES "^QT_")
       # filter out the very long list of Qt libraries and include dirs
