commit 4d4546680f1b8cb7397d34f88fff4e8d4ec4a913
Author: David Rohr <drohr@jwdt.org>
Date:   Tue Jul 26 08:45:07 2022 +0200

    Replace deprecated boost:filesystem::ifstream/ofstream with std::

diff --git a/dds-agent/src/CommanderChannel.cpp b/dds-agent/src/CommanderChannel.cpp
index 0a9b809..bb1ac49 100644
--- a/dds-agent/src/CommanderChannel.cpp
+++ b/dds-agent/src/CommanderChannel.cpp
@@ -664,7 +664,7 @@ bool CCommanderChannel::on_cmdACTIVATE_USER_TASK(SCommandAttachmentImpl<cmdACTIV
         const fs::path pathTaskWrapper(pathSlotDir / "dds_user_task_wrapper.sh");

         // Replace placeholders in the task wrapper
-        fs::ifstream fTaskWrapper(pathTaskWrapperIn.native());
+        std::ifstream fTaskWrapper(pathTaskWrapperIn.native());
         if (!fTaskWrapper.is_open())
             throw runtime_error("Failed to open task wrapper template.");
         string sTaskWrapperContent((istreambuf_iterator<char>(fTaskWrapper)), istreambuf_iterator<char>());
@@ -676,7 +676,7 @@ bool CCommanderChannel::on_cmdACTIVATE_USER_TASK(SCommandAttachmentImpl<cmdACTIV
         // JOB SCRIPT --- Task Executable
         boost::replace_all(sTaskWrapperContent, "# %DDS_USER_TASK%", sUsrExe);

-        fs::ofstream fTaskWrapperOut(pathTaskWrapper.native());
+        std::ofstream fTaskWrapperOut(pathTaskWrapper.native());
         if (!fTaskWrapperOut.is_open())
             throw runtime_error("Failed to create task wrapper script.");

