--- root/build_for_fair/lib/root/ROOT.py
+++ root/build_for_fair/lib/root/ROOT.py
@@ -83,6 +83,12 @@
 
    readline.parse_and_bind( 'tab: complete' )
    readline.parse_and_bind( 'set show-all-if-ambiguous On' )
+# history file 
+   histfile = os.path.join(os.environ['HOME'], '.pythonhistory') 
+   try: 
+    readline.read_history_file(histfile) 
+   except IOError: 
+    pass 
 except:
  # module readline typically doesn't exist on non-Unix platforms
    pass
