--- cmake/modules/FindXROOTD.cmake
+++ cmake/modules/FindXROOTD.cmake
@@ -21,6 +21,7 @@ set(searchpath ${XROOTD_ROOT_DIR} $ENV{XROOTD_ROOT_DIR} $ENV{XRDSYS} /opt/xrootd
 find_path(XROOTD_INCLUDE_DIR NAMES XrdVersion.hh
   HINTS ${searchpath}
   PATH_SUFFIXES include include/xrootd
+  NO_DEFAULT_PATH
 )
 
 if (XROOTD_INCLUDE_DIR)
@@ -81,7 +82,9 @@ if(XROOTD_FOUND)
       find_library(XROOTD_${l}_LIBRARY
          NAMES ${l} 
          HINTS ${searchpath}
-         PATH_SUFFIXES lib lib64)
+         PATH_SUFFIXES lib lib64
+         NO_DEFAULT_PATH
+         )
       list(APPEND XROOTD_LIBRARIES ${XROOTD_${l}_LIBRARY})
     endforeach()
 
@@ -89,7 +92,9 @@ if(XROOTD_FOUND)
       find_library(XROOTD_XrdNetUtil_LIBRARY
         NAMES XrdNetUtil
         HINTS ${searchpath}
-        PATH_SUFFIXES lib lib64)
+        PATH_SUFFIXES lib lib64
+        NO_DEFAULT_PATH
+        )
       list(APPEND XROOTD_LIBRARIES ${XROOTD_XrdNetUtil_LIBRARY})
     endif ()
   else()
@@ -98,7 +103,9 @@ if(XROOTD_FOUND)
     find_library(XROOTD_XrdMain_LIBRARY
        NAMES XrdMain
        HINTS ${searchpath}
-       PATH_SUFFIXES lib lib64)
+       PATH_SUFFIXES lib lib64
+       NO_DEFAULT_PATH
+       )
     if (XROOTD_XrdMain_LIBRARY)
        list(APPEND XROOTD_LIBRARIES ${XROOTD_XrdMain_LIBRARY})
     else ()
@@ -112,7 +119,9 @@ if(XROOTD_FOUND)
     find_library(XROOTD_XrdUtils_LIBRARY
        NAMES XrdUtils
        HINTS ${searchpath}
-       PATH_SUFFIXES lib lib64)
+       PATH_SUFFIXES lib lib64
+       NO_DEFAULT_PATH
+       )
     if (XROOTD_XrdUtils_LIBRARY)
        list(APPEND XROOTD_LIBRARIES ${XROOTD_XrdUtils_LIBRARY})
     endif ()
@@ -121,7 +130,9 @@ if(XROOTD_FOUND)
     find_library(XROOTD_XrdClient_LIBRARY
        NAMES XrdClient
        HINTS ${searchpath}
-       PATH_SUFFIXES lib lib64)
+       PATH_SUFFIXES lib lib64
+       NO_DEFAULT_PATH
+       )
     if (XROOTD_XrdClient_LIBRARY)
        list(APPEND XROOTD_LIBRARIES ${XROOTD_XrdClient_LIBRARY})
     else ()
@@ -136,7 +147,9 @@ if(XROOTD_FOUND)
        find_library(XROOTD_XrdCl_LIBRARY
           NAMES XrdCl
           HINTS ${searchpath}
-          PATH_SUFFIXES lib lib64)
+          PATH_SUFFIXES lib lib64
+          NO_DEFAULT_PATH
+          )
        if (XROOTD_XrdCl_LIBRARY)
           list(APPEND XROOTD_LIBRARIES ${XROOTD_XrdCl_LIBRARY})
        endif ()
-- 
2.2.1

