--- spack-src/CMakeLists.txt
+++ spack-src/CMakeLists.txt
@@ -90,6 +90,9 @@
     find_package2(PUBLIC ${dep} REQUIRED VERSION ${FairLogger_${dep}_VERSION})
   endif()
 endforeach()
+get_target_property(fairlogger_include FairLogger::FairLogger INTERFACE_INCLUDE_DIRECTORIES)
+set(r3broot_deps_includes ${r3broot_deps_includes} ${fairlogger_include})
+
 find_package2(PUBLIC Pythia6)
 find_package2(PUBLIC Pythia8)
 
--- spack-src/field/CMakeLists.txt
+++ spack-src/field/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/r3bbase/CMakeLists.txt
+++ spack-src/r3bbase/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 Set(R3BSOF_SOURCE_DIR ${R3BROOT_SOURCE_DIR}/sofia)
 
--- spack-src/passive/CMakeLists.txt
+++ spack-src/passive/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/califa/CMakeLists.txt
+++ spack-src/califa/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 ${Boost_INCLUDE_DIR}
 )
 
--- spack-src/xball/CMakeLists.txt
+++ spack-src/xball/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/dch/CMakeLists.txt
+++ spack-src/dch/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/gfi/CMakeLists.txt
+++ spack-src/gfi/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/fi3a/CMakeLists.txt
+++ spack-src/fi3a/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/fi3b/CMakeLists.txt
+++ spack-src/fi3b/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/fi4/CMakeLists.txt
+++ spack-src/fi4/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/fi5/CMakeLists.txt
+++ spack-src/fi5/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/fi6/CMakeLists.txt
+++ spack-src/fi6/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/fi7/CMakeLists.txt
+++ spack-src/fi7/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/fi8/CMakeLists.txt
+++ spack-src/fi8/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/fi10/CMakeLists.txt
+++ spack-src/fi10/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/fi11/CMakeLists.txt
+++ spack-src/fi11/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/neuland/digitizing/CMakeLists.txt
+++ spack-src/neuland/digitizing/CMakeLists.txt
@@ -11,6 +11,12 @@
 # or submit itself to any jurisdiction.                                      #
 ##############################################################################
 
+Set(SYSTEM_INCLUDE_DIRECTORIES 
+${SYSTEM_INCLUDE_DIRECTORIES}
+${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
+)
+
 set(LIBRARY_NAME R3BNeulandDigitizing)
 set(LINKDEF NeulandDigitizingLinkDef.h)
 
--- spack-src/passive/CMakeLists.txt
+++ spack-src/passive/CMakeLists.txt
@@ -18,14 +18,15 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
-${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
 ${R3BROOT_SOURCE_DIR}/r3bbase
 ${R3BROOT_SOURCE_DIR}/field
 ${R3BROOT_SOURCE_DIR}/passive
-${R3BROOT_SOURCE_DIR}/tracking )
+${R3BROOT_SOURCE_DIR}/tracking
+${r3broot_deps_includes}
+)
 
 include_directories( ${INCLUDE_DIRECTORIES})
 include_directories(SYSTEM ${SYSTEM_INCLUDE_DIRECTORIES})
--- spack-src/neuland/simulation/CMakeLists.txt
+++ spack-src/neuland/simulation/CMakeLists.txt
@@ -11,6 +11,12 @@
 # or submit itself to any jurisdiction.                                      #
 ##############################################################################
 
+Set(SYSTEM_INCLUDE_DIRECTORIES 
+${SYSTEM_INCLUDE_DIRECTORIES}
+${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
+)
+
 set(LIBRARY_NAME R3BNeulandSimulation)
 set(LINKDEF NeulandSimulationLinkDef.h)
 
--- spack-src/mtof/CMakeLists.txt
+++ spack-src/mtof/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/tof/CMakeLists.txt
+++ spack-src/tof/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/tracker/CMakeLists.txt
+++ spack-src/tracker/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/music/CMakeLists.txt
+++ spack-src/music/CMakeLists.txt
@@ -15,7 +15,10 @@
 # the array .
 # The extension is already found.  Any number of sources could be listed here.
 
-Set(SYSTEM_INCLUDE_DIRECTORIES ${SYSTEM_INCLUDE_DIRECTORIES} ${BASE_INCLUDE_DIRECTORIES} )
+Set(SYSTEM_INCLUDE_DIRECTORIES ${SYSTEM_INCLUDE_DIRECTORIES}
+${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
+)
 
 if(EXISTS "${R3BROOT_SOURCE_DIR}/sofia/")
 set(WITH_SOFIA true)
--- spack-src/sfi/CMakeLists.txt
+++ spack-src/sfi/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/land/CMakeLists.txt
+++ spack-src/land/CMakeLists.txt
@@ -14,7 +14,10 @@
 # Create a library called "libR3BCal" which includes the source files given in the array . The
 # extension is already found.  Any number of sources could be listed here.
 
-set(SYSTEM_INCLUDE_DIRECTORIES ${SYSTEM_INCLUDE_DIRECTORIES} ${BASE_INCLUDE_DIRECTORIES})
+set(SYSTEM_INCLUDE_DIRECTORIES ${SYSTEM_INCLUDE_DIRECTORIES}
+${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
+)
 
 set(INCLUDE_DIRECTORIES
     # put here all directories where header files are located
--- spack-src/mfi/CMakeLists.txt
+++ spack-src/mfi/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/startrack/CMakeLists.txt
+++ spack-src/startrack/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/psp/CMakeLists.txt
+++ spack-src/psp/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/fi12/CMakeLists.txt
+++ spack-src/fi12/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/fi13/CMakeLists.txt
+++ spack-src/fi13/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
--- spack-src/lumon/CMakeLists.txt
+++ spack-src/lumon/CMakeLists.txt
@@ -18,6 +18,7 @@
 Set(SYSTEM_INCLUDE_DIRECTORIES 
 ${SYSTEM_INCLUDE_DIRECTORIES}
 ${BASE_INCLUDE_DIRECTORIES}
+${r3broot_deps_includes}
 )
 
 set(INCLUDE_DIRECTORIES
